engine/gameobject/src/gameobject/gameobject.h

```diff
@@ -173,6 +173,8 @@ namespace dmGameObject
         int32_t  m_TouchCount;
         char     m_Text[dmHID::MAX_CHAR_COUNT];
         uint32_t m_TextCount;
+        char     m_GamepadName[dmHID::MAX_CHAR_COUNT];
+        uint32_t m_GamepadNameCount;
```

Comment:
tbh, I vote for noth having this "count".
I think it's perfectly fine to call strlen() once it's needed (which is very rare), as opposed to increasing the size of this struct.
E.g. this struct should probably have at least some union members...


Also, due to the (in my opinion misnamed m_TextCount, which should be TextLength or TextSize):

In our code, "count" refers to number of items. E.g. number of touches.
"Size" refers to the size (in bytes). "Length" is length (in units).
And here "count" is ambigous. Perhaps it means number of game pad names?

Reply:
Come to think of it. Since the max char count is so large (256 bytes) and the event cannot be a text event and a gamepad event at the same time, you can reuse the m_Text field.

It only needs to be commented properly in the struct.

Reply:
@JCash I did reuse the field and tried to be cautious about it. I will do some more testing and remove the draft if everything goes well.

Reply:
@JCash I think you can re-review now :) I did more testing on Android and added comments about the results to the description.